juju/juju
# All space infos

## Description of change

This lifts the `MapWithSpaceNames` requirement of state out of the function. Instead, it requires a lookup of `network.SpaceInfos` to determine if and how to map ProviderIDs to SpaceNames.

The great thing with this is that we can now prevent excessive calls to state when requesting the space names for the bindings.

Future work should attempt to lift this from out of state to make in conceptually pure. This has no requirements on state, it transforms one piece of data from one to another, and isn't relatable to what's in the database.

## QA steps

Ensure that the tests pass. This is mainly mechanical change, but existing tests should work and pass as previously had done.
